### PR TITLE
nixos/paperless: add environmentFile option

### PR DESCRIPTION
**Pull request by bct**
Adds an `environmentFile` option to the `paperless` module.

This allows arbitrary secrets to be passed to Paperless as environment variables. This is useful for the settings `PAPERLESS_DBPASS` and `PAPERLESS_EMAIL_HOST_PASSWORD`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] `nix-build -A nixosTests.paperless`
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by leona-ya — CHANGES_REQUESTED** [review]
I like the feature, but I'm not really happy with the naming because it doesn't directly imply that it's for environment variables. In almost all other services such an option is called `environmentFile`. If you're really unhappy with that name, I can live with other names if they have something like env/environment in them.

File: nixos/modules/services/misc/paperless.nix [hunk 1]
```diff
@@ -34,9 +34,14 @@ let
...
 
-  manage = pkgs.writeShellScript "manage" ''
+  manage = let
+    maybeSourceSecretsFile = if cfg.secretsFile != null
+      then "source ${cfg.secretsFile}"
+      else "";
+  in pkgs.writeShellScript "manage" ''
     set -o allexport # Export the following env vars
     ${lib.toShellVars env}
+    ${maybeSourceSecretsFile}
```
Comment (SuperSandro2000) [inline nixos/modules/services/misc/paperless.nix:44]:
```suggestion
  manage = pkgs.writeShellScript "manage" ''
    set -o allexport # Export the following env vars
    ${lib.toShellVars env}
    ${lib.optionalString (cfg.secretsFile != null) "source ${cfg.secretsFile}"}
```

**Comment by bct** [comment]
OK, changes made.

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Review by leona-ya — APPROVED** [review]

**Comment by Mj-q8** [comment]
?

File: nixos/modules/services/misc/paperless.nix [hunk 2]
```diff
@@ -228,6 +234,27 @@ in
...
+
+    secretsFile = mkOption {
+      type = types.nullOr (
+        types.str
+        // {
+          # We don't want users to be able to pass a path literal here but
+          # it should look like a path.
+          check = it: lib.isString it && lib.types.path.check it;
+        }
+      );
```
[the remark this answers is not in the document]
Comment (jfly) [reply nixos/modules/services/misc/paperless.nix]: For the record, I've put out a PR to add a `types.pathNotInStore` over here: https://github.com/NixOS/nixpkgs/pull/373287

**Pull request merged**